tests/python/PythonPackageInitBearTest.py

```diff
@@ -0,0 +1,61 @@
+import unittest
+import os
+
+from queue import Queue
+
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+
+from bears.python.PythonPackageInitBear import PythonPackageInitBear
+
+
+class PythonPackageInitBearTest(unittest.TestCase):
+
+    def setUp(self):
+        
+        os.mkdir('with-init')
+
+
+        open('with-init/__init__.py', 'a').close()
+
+        os.mkdir('ignored-no-init')
+        open('ignored-no-init/ignored.py', 'a').close()
+
+        os.mkdir('not-ignored-no-init')
+        open('not-ignored-no-init/test-file.py', 'a').close()
+
+        self.queue = Queue()
+        self.file_dict = {}
+        self.section = Section('PythonPackageInit')
+        self.section.add_or_create_setting(Setting('ignored_dirs', 'ignored-no-init'))
+
+        self.uut = PythonPackageInitBear(self.file_dict, self.section,
+                                         self.queue)
+
+    def tearDown(self):
+        os.remove('with-init/__init__.py')
+        os.remove('ignored-no-init/ignored.py')
+        os.remove('not-ignored-no-init/test-file.py')
+        os.rmdir('with-init')
+        os.rmdir('ignored-no-init')
+        os.rmdir('not-ignored-no-init')
+
+    def run_uut(self, *args, **kwargs):
+        return list(result.message for result in self.uut.run())
+
+    def test_init_creation(self):
+        self.file_dict = {os.path.abspath('not-ignored-no-init/test-file.py'):
```

Comment:
This is the line that caused all this! Instead of `self.file_dict` you need to use `self.uut.file_dict`. This is because python is always pass by value and you set uut's file_dict to `{}` in the setup blob. Thus, here all you're modifying is the test class's file_dict and that doesn't affect what uut sees. instead, you need to modify uut's file dict. (Sorry if you knew this, I'm writing this partly for myself too)
Thanks for the mental workout!